internal/store/leaf/fsck.go

```diff
@@ -27,6 +27,67 @@ const (
 	errsFatal                  = 2 // an error that terminated the function early
 )
 
+func (e ErrorSeverity) String() string {
+	switch {
+	case e == errsNonFatal:
+		return "non-fatal"
+	case e == errsFatal:
+		return "fatal"
+	default:
+		return "nil"
+	}
+}
+
+type fsckMultiError struct {
+	Severity ErrorSeverity
+	Errors   []error
+}
+
+func (f *fsckMultiError) IsError() bool {
+	return len(f.Errors) > 0
+}
+
+func (f *fsckMultiError) Error() string {
+	if len(f.Errors) < 1 {
+		return ""
+	}
+
+	var sb strings.Builder
+	fmt.Fprintf(&sb, "[%s] ", f.Severity.String())
+	msgs := make([]string, 0, len(f.Errors))
+	for _, e := range f.Errors {
+		msgs = append(msgs, e.Error())
+	}
+	sb.WriteString(strings.Join(msgs, ", "))
```

Comment:
Wouldn't it be best to wrap errors using `%w` and `Fprintf` instead?

```suggestion
	addComma := false
	for _, e := range f.Errors {
		if addComma {
			fmt.Fprint(%sb, ", ")
		}
		fmt.Fprintf(%sb, "%w", e.Error())
		addComma = true
	}
```

Reply:
I don't think `%w` is valid outside of `fmt.Errorf`.